packages/tradershub/src/hooks/useAccountSwitcher.ts

```diff
@@ -26,15 +26,22 @@ const useAccountSwitcher = () => {
     const [selectedAccount, setSelected] = useState(activeType);
     const firstRealLoginId = tradingAccountsList?.find(acc => !acc.is_virtual)?.loginid;
     const demoLoginId = tradingAccountsList?.find(acc => acc.is_virtual)?.loginid;
+    const { data: isDIEL } = useIsDIELEnabled();
 
     useEffect(() => {
+        if (isDIEL && activeAccountType === 'demo') {
+            setUIState({
+                regulation: 'Non-EU',
+            });
+        }
```

Comment:
What is this logic? not sure what are we supposed to get here?

Reply:
We are setting the ui state for the components to know. Instead of going to every component and telling them now its Non-EU. This is because the states are being persisted in demo as well.

Reply:
Well I was hoping that we won't hard-code anything in there!

Reply:
Will change it boss